src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs

```diff
@@ -55,9 +55,8 @@ public static partial class PlatformDetection
         public static bool IsNotFedoraOrRedHatFamily => !IsFedora && !IsRedHatFamily;
         public static bool IsNotDebian10 => !IsDebian10;
 
-        public static bool IsSuperUser => IsBrowser || IsWindows ? false : libc.geteuid() == 0;
-
-        public static bool IsUnixAndSuperUser => !IsWindows && IsSuperUser;
+        public static bool IsSuperUser => IsBrowser || IsWindows ? false : IsPrivilegedProcess;
+        public static bool IsUnixAndSuperUser => !IsWindows && IsPrivilegedProcess;
```

Comment:
Do we still need these? (Same question for the helpers in the Windows.cs file).

My expectation was that all tests could just use e.g.
```C#
[ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsPrivilegedProcess))]
```
and if they also needed to only run on a specific platform (or be excluded from a specific platform), also use the standard:
```C#
[PlatformSpecific(...)]
```
mechanism.

Also, if the `IsBrowser` guard is necessary, that brings into question "why", and whether that means there's something wrong with IsPrivilegedProcess in the browser.

Reply:
Thanks for clarifying what you were expecting--a deeper cut to remove the platform-specific helpers entirely, not just to refactor them. I couldn't see any evidence of why `IsBrowser` was necessary, but I anticipate some checks will fail and the reasoning will be revealed. If that does occur, I can then add `[SkipOnPlatform]` attributes where appropriate.